### PR TITLE
CH4 Coverity Fixes

### PR DESCRIPTION
**Pull request by wesbland**

### CODE REVIEW

**Comment by wesbland** [comment]
Sorry about the placeholder title. Somehow our internal PR template leaked out when I created this in `hub`. No secrets there though. 😄

**Comment by wesbland** [comment]
test:jenkins/ch4/most

File: src/mpid/ch4/src/ch4_globals.c [hunk 1]
```diff
@@ -95,7 +95,7 @@ int MPID_Abort(MPIR_Comm * comm, int mpi_errno, int exit_code, const char *error
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_ABORT);
     fflush(stderr);
     fflush(stdout);
-    if (MPIR_Comm_size(comm) == 1)
+    if (comm && MPIR_Comm_size(comm) == 1)
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4_globals.c]: Do we want to use `||` instead to limit the scope?

Comment (wesbland) [reply src/mpid/ch4/src/ch4_globals.c]: The complaint from Coverity was that `comm` might be `NULL`, so `||` wouldn't solve that problem.

Comment (raffenet) [reply src/mpid/ch4/src/ch4_globals.c]: How about `if (comm == NULL || MPIR_COMM_size(comm) == 1)`. Evaluating left to right should guarantee we don't dereference `NULL`.

Comment (wesbland) [reply src/mpid/ch4/src/ch4_globals.c]: Yeah, that's probably fine. I think the point here was that if there's only one process, it shouldn't call `PMI_Abort`. Maybe it should check `MPIR_Comm_size(MPIR_Process.comm_world) == 1`?

Comment (raffenet) [reply src/mpid/ch4/src/ch4_globals.c]: `PMI_Abort` has no scope capabilities. Think of my proposal as effectively aborting only `MPI_COMM_SELF` when there is no associated comm/window/file.

Comment (wesbland) [reply src/mpid/ch4/src/ch4_globals.c]: Done.

**Comment by wesbland** [comment]
test:jenkins/ch4/most

**Comment by wesbland** [comment]
@hajimefu, can you validate the change in the last patch?

**Comment by hajimefu** [comment]
@shawnccx Can you also review the "ch4/ofi: Push endpoint at back in error case" patch?

**Comment by shawnccx** [comment]
@hajimefu looks good

**Comment by raffenet** [comment]
Looks good now. I tweaked the subject line of the last commit based on my reading of the code. Does it make sense?

**Comment by hajimefu** [comment]
"ch4/ofi: *Correctly* release transmit context in error case"?

**Comment by wesbland** [comment]
Either is fine with me. I don't care.

**Comment by raffenet** [comment]
K, I'll change it to Hajime's version and merge.

**Pull request merged**